compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/LocalDeclarationsLowering.kt

```diff
@@ -726,6 +726,8 @@ class LocalDeclarationsLowering(
             )
             newDeclaration.recordTransformedValueParameters(constructorContext)
 
+            newDeclaration.annotations = oldDeclaration.annotations
```

Comment:
Am I correct that the old declaration is dropped?

Reply:
Yes, the old declaration is dropped. We are hoisting out the class and replacing the constructor with a new one.

Reply:
Thanks for clarification